### DISCUSSION TITLE
Maven profile usage

### DISCUSSION THREAD

**Discussion by wnleao** [sevdokimov/log-viewer#33] [General]
What is the expected behavior when building with "demo-projects" profile? Is it expected to also build the frontend? Currently, the frontend is being built in both existing profiles.

**Comment by sevdokimov** [comment]
"demo-projects" and "build-frontend" are independent. 
"demo-projects" allows enabling/disabling demo projects. I don't want to publish demo-projects to maven repository.  
"build-frontend" allows making a build without rebuilding UI. Building UI takes a lot of time. I use it when I made a change in Java only and UI is already compiled.   "build-frontend" is used to optimize development process only.
